### ISSUE TITLE
Config web ui

### ISSUE THREAD

**Issue by deontologician** [rethinkdb/horizon#154]
There needs to be a user-friendly web ui for horizon app configuration. Anything in the config file ( #153 ) should be possible to modify through the ui.

We talked about having the ui served with `--dev` on a different port, but we should probably mount it at `/horizon/admin` or something. It would modify the config file on disk directly. I don't know how we want the endpoint to work. It should be out of band from the horizon client, so maybe a rest endpoint.

pinging @Tryneus and @dalanmiller for any stuff I've forgotten. Also pinging @mglukhovsky since he'll probably be building the frontend for this

**Comment by casoetan** [comment]
This looks fantastic.

Any updates on this?

**Comment by dim882** [comment]
Is it possible to use the regular rethinkdb web ui in the meantime while this is in progress?

**Comment by deontologician** [comment]
yes, when you do 'hz serve' it will show you the port number for your
rethinkdb admin ui

**Comment by segphault** [comment]
@casoetan: This is still a work in progress. I got past the major architectural hurdles that I described earlier in the thread. My current implementation now actively tracks each horizon instance and can automatically cull records from instances that are no longer sending a heartbeat to the database. I'm currently working on adding support for managing permissions and groups. I don't have any realistic timeline for when this will be available, but it's moving forward.

@mrdanimal: yes, you can use RethinkDB's standard admin UI to manage the underlying tables that Horizon uses. The data explorer, sharding/replication, table management, etc all work exactly as expected.

**Comment by lavvy** [comment]
Any update?  Is there a feature of adding custom reql queries through the web ui

**Comment by jhicken** [comment]
@segphault are you working on a branch? or a separate project? Can I help? Is there a list of outstanding tasks?

**Comment by segphault** [comment]
I'm not actively working on this feature at the moment. Due to all of the code churn in Horizon, I've had to rewrite the dashboard's backend integration code three times now. I've been waiting for Horizon plugins to land, because I believe that the plugin system will make it dramatically easier to build and maintain the instrumentation and custom queries that are needed to support this kind of dashboard.

You can find the current proof-of-concept implementation in the ryan_dashboard branch. It's very rough right now and will definitely not be adaptable to current or future versions of Horizon.

I'm currently prioritizing my work on Basejump, but I might revisit the Horizon dashboard at some point after plugins land. If others want to take on this project, I'd strongly encourage you to wait for plugins to land. While some of the queries and logic in my branch might prove instructive, you probably won't be able to repurpose much of the existing code.

**Comment by jhicken** [comment]
@segphault Thanks thats a very helpful update.

On a side note I saw your presentation on basejump. I am super interested in it. But I looks like you have only made the router public. Is the UI not public for the same reasons you mentioned above? specifically... the churn? I would also be interested in pitching in on that ui.

Sorry everyone for kinda veering off topic.

**Comment by emeagenciadigital** [comment]
is this dead?
